cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/sql/dataset/SQLDataset.java

```diff
@@ -14,14 +14,16 @@
  * the License.
  */
 
-package io.cdap.cdap.etl.engine;
+package io.cdap.cdap.etl.api.engine.sql.dataset;
+
+import io.cdap.cdap.api.data.schema.Schema;
 
 /**
- * Status for SQL Engine Jobs.
+ * Represents a dataset that lives in a SQL engine outside of spark
+ * @param <T> the type for the records stored in the SQL Engine.
  */
-public enum SQLEngineJobStatus {
-  CREATED,
-  RUNNING,
-  COMPLETED,
-  FAILED
+public interface SQLDataset<T> {
+  String getDatasetName();
```

Comment:
I don't think you need getDatasetName. Dataset may be not materialized at all if it passed to another join and combined into a single statement

Reply:
While I'm not using the Dataset name at this time, I think it is still useful to have this information in the SQL Dataset. Since this matches the Stage name in CDAP, this can be used for logging/debugging purposes shall the need arise within the SQL engine.

Reply:
When we get to the point of combining operations, the name of the dataset would be the name of the "last" operation. If A->B->C can be combined into one step, the result of this operation is C, as any following stages will use the output of C as the input to their respective stages.